securing_apps/topics/cli-apps/cli-apps.adoc

```diff
@@ -0,0 +1,137 @@
+[[_kcinit]]
+
+== Securing Command Line Applications
+
+Many applications need a command line interface to administer and configure themselves.
+If these command line interfaces are invoking on back-end admin REST interfaces that
+are secured by {project_name}, they need a way to obtain a token to invoke on them.
+This chapter discusses some utilities that {project_name} has that you can
+use to secure your command line interfaces.
+
+The initial and standard way to obtain a token for a command line application
+is to prompt the user for username and password and invoke on the <<_resource_owner_password_credentials_flow, Resource Owners Credential Grant>>
+ OAuth endpoint that {project_name} has.  While a completely acceptable solution,
+there are some disadvantages to this approach.  The obvious one is that you'll have to
+code the protocol and manage the tokens.  This means your CLI app will have to manage
+and be aware of token timeouts and refresh tokens.  The second disadvantage is
+that if there are other CLI apps using {project_name}, you'll have to relogin
+to each different CLI app that is secured by {project_name}.  No SSO on the command line.
+Finally, your app will have to be hard coded to specific credential types.  If you've
```

Comment:
The second disadvantage is that if there are other CLI apps using {project_name}, you will have to login multiple times, once for each CLI app that is secured by {project_name} as there is no SSO available on the command line.

Reply:
nice.